From 4356779fe843fb30b3728f5fdab049c7291d89c6 Mon Sep 17 00:00:00 2001
From: "Field G. Van Zee" <field@cs.utexas.edu>
Date: Mon, 23 Sep 2019 15:15:15 -0500
Subject: [PATCH] Set execute bits of shared library at install-time.

Details:
- Modified the 0644 octal code used during installation of shared
  libraries to 0755. Thanks to Adam J. Stewart for reporting this
  issue in #24.
---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index d856af4c..6bd8bdce 100644
--- a/Makefile
+++ b/Makefile
@@ -643,11 +643,11 @@ endif
 $(INSTALL_LIBDIR)/%.$(LIBFLAME_SO_MMB_EXT): $(BASE_LIB_PATH)/%.$(SHLIB_EXT) $(CONFIG_MK_FILE)
 ifeq ($(ENABLE_VERBOSE),yes)
 	$(MKDIR) $(@D)
-	$(INSTALL) -m 0644 $< $@
+	$(INSTALL) -m 0755 $< $@
 else
 	@echo "Installing $(@F) into $(INSTALL_LIBDIR)/"
 	@$(MKDIR) $(@D)
-	@$(INSTALL) -m 0644 $< $@
+	@$(INSTALL) -m 0755 $< $@
 endif
 
 
